File: src/FSharpLab1/Problem13.fs
module Problem13

open System.IO

let numbersPath = 
    Path.Combine(__SOURCE_DIRECTORY__, "..", "..", "resources", "0013_numbers.txt")


let getNumbers path =
    File.ReadAllLines path 
    |> Array.map (fun line -> bigint.Parse(line.Trim()))


let recursive () =
    let numbers = getNumbers numbersPath
    let sum = numbers |> Array.sum
    sum.ToString().Substring(0, 10) |> int64


let modular () =
    numbersPath
    |> getNumbers
    |> Array.sum
    |> string
    |> fun s -> s.Substring(0, 10)
    |> int64

let solutions = [ recursive; modular ]

File: other/problem13.py
def solve():
    with open('Lab1/resources/0013_numbers.txt', 'r') as file:
        numbers = [int(line.strip()) for line in file]
    total_sum = sum(numbers)
    return str(total_sum)[:10]

if __name__ == "__main__":
    print(solve())